### PR TITLE
new coverage

### PR DESCRIPTION
**Pull request by noahsw**
better fix for #218

### CODE REVIEW

**Comment by noahsw** [comment]
hey @serut, I have a clean PR now on top of Meteor 1.4. CI is hitting this error:

```
phantomjs: TypeError: undefined is not an object (evaluating 'Package['service-configuration'].ServiceConfiguration')
```

When running locally, some of my full-app tests are failing without helpful stacktraces. Could be a local timing thing so I'd trust the CI results more.

**Comment by serut** [comment]
Can you try to spam the CI platform to see how many times it happens ? 
Like I did there https://github.com/serut/meteor-coverage-app-exemple/blob/beaa76fdc5516f909ec79f044db276cacd241a2b/circle.yml

I think you have 0.5% chance that a dependency is corrumpted, but your project is large enought to see it more often.

**Comment by noahsw** [comment]
Arg - you're right. I just got a pass. I'll rebuild it a couple more times to see what my success rate is. Damn that's annoying... Any idea what the culprit is? I've seen this happen before I added your coverage library too. A rebuild almost always fixes it.

**Comment by serut** [comment]
On my reliability test with 200x coverage tests and 200x without coverage, I had 0 to 10 % of fail with coverage activated and 0% without. And that was with meteor-coverage 0.9 and not 1.1, the last version instruments a lot less files, so it reduces the chance to fail.

**Comment by noahsw** [comment]
Unfortunately I'm only 2 for 8 success rate. https://circleci.com/gh/winsleague/winsleague/tree/feature%2F218-coverage-clean

I used to be 9 for 10 at least.

**Comment by serut** [comment]
I reread your log. There is only 1 corrumpted dependency that can be related to coverage ! 
https://circleci.com/gh/winsleague/winsleague/793

# Remaining errors
```
C: Full-app test of Pools Full-app test of Pools.show "before each" hook for "should allow user to switch seasons" : failed
  http://localhost:4096/app/app.js?hash=c1bd626b6ff5af2ff1e015d07fb339fdbee232a9:3745:58
http://localhost:4096/packages/meteor.js?hash=ae8b8affa9680bf9720bd8f7fa112f13a62f71c3:1105:27
tryCallOne@http://localhost:4096/packages/promise.js?hash=2d5b025794b9c9e54e5cf6bea71c642ef921a2ed:277:14
http://localhost:4096/packages/promise.js?hash=2d5b025794b9c9e54e5cf6bea71c642ef921a2ed:363:25
flush@http://localhost:4096/packages/promise.js?hash=2d5b025794b9c9e54e5cf6bea71c642ef921a2ed:514:33
run@http://localhost:4096/packages/modules.js?hash=3120e85473dba9c399fc87843ec8640ffe94e47a:130845:19
drainQueue@http://localhost:4096/packages/modules.js?hash=3120e85473dba9c399fc87843ec8640ffe94e47a:130815:45
```
https://circleci.com/gh/winsleague/winsleague/798, https://circleci.com/gh/winsleague/winsleague/795


----------

```
S: MLB Game Data "before each" hook: "before each" hook: wrappedFunction for "should ingest all games for on June 15, 2016" : failed
  Error: timeout of 10000ms exceeded. Ensure the done() callback is being called in this test.
```
timeout can be handle. Just set this line in your test with the following : https://github.com/serut/meteor-coverage/blob/8a46f5ac1b038ebcb108db36865f4456a5c254ff/client/methods.e2e.tests.js#L23  
https://circleci.com/gh/winsleague/winsleague/796
https://circleci.com/gh/winsleague/winsleague/794
https://circleci.com/gh/winsleague/winsleague/791

**Comment by noahsw** [comment]
@serut I spent some time doing some refactoring and things seem to be far more stable! Thx for your help.

It's really cool that the HTML reports are available as CI artifacts as well. One minor complaint is the CSS and link paths used in the HTML report are absolute instead of relative and therefore break on CircleCI. See https://circleci.com/gh/winsleague/winsleague/809 and https://809-33826277-gh.circle-artifacts.com/0/home/ubuntu/winsleague/app/.coverage/index.html. Any idea how to fix?

If this passes a couple more times, I'll merge!

**Comment by noahsw** [comment]
@serut when my tests run, they're far more reliable. however, I'm hitting the Package undefined error ~20% of the time. will try to stabilize my other tests so we can focus on this...

**Comment by noahsw** [comment]
@serut - just an update. I made good progress stabilizing the tests, but then started getting more failures on my acceptance tests with chimp. To isolate the impact of code coverage, I created a separate branch (feature/stable-tests) to merge my stability improvements into master. Now this branch truly only represents the code coverage changes.

I'm not sure why the code coverage should impact acceptance tests though. I'm running them with COVERAGE=0. Are we sure there's absolutely no instrumentation going on in this case?

**Comment by serut** [comment]
Yes, when coverage is not active, there is no specific impact. 

If I'm right, Chimp runs the real meteor application and then launch a test runner ? In that case, meteor-coverage is not included in your app because it has a `debugOnly` flag in the `package.js` file

**Comment by noahsw** [comment]
I believe so but I'm not a Chimp expert. Here's how I'm calling Chimp: https://github.com/winsleague/winsleague/blob/8766775037222fd399bdb846d341da828bac24d8/app/.testing/chimp.js

I'm a little suspicious that coverage is totally turned off because:
a) I see it mentioned in the logs of https://circleci.com/gh/winsleague/winsleague/862 (although it could just be because of https://github.com/serut/meteor-coverage/issues/43)
b) You can see the last few runs on master and some of my feature branches have passed just fine (since my test refactoring too). All this branch does is turn on coverage and all of the sudden my acceptance tests start failing. You can see I'm trying to bump up the timeouts in this branch but that doesn't seem to help. I've also tried to explicitly pass COVERAGE=0 to meteor as well but that isn't making a difference.

How can we 100% make sure coverage is off? Maybe if meteor-coverage is part of the build, it should always log out its state?

**Comment by serut** [comment]
In that case add the `COVERAGE_VERBOSE=1`.

I was wrong, https://github.com/winsleague/winsleague/blob/8766775037222fd399bdb846d341da828bac24d8/app/.testing/chimp.js#L13 is using a `meteor test` command so meteor-coverage is included in the build.

To sum up, even if meteor-coverage is not active, it creates some timeout on your IT ?

**Comment by noahsw** [comment]
I added `COVERAGE_VERBOSE=1` and it looks like everything is disabled. But the tests still fail 100% of the time. Super weird. I can remove some of that extra timeout code to make it completely apples to apples but I doubt it will have an impact.

File: app/.meteor/versions [hunk 1]
```diff
@@ -64,7 +64,7 @@ lai:collection-extensions@0.1.4
 launch-screen@1.0.12
 less@2.6.5
 livedata@1.0.18
-lmieulet:meteor-coverage@0.9.4
+lmieulet:meteor-coverage@1.1.3
```
Comment (serut) [inline app/.meteor/versions]: Can be updated

**Pull request closed without merging**